### PR TITLE
Update Table Row to FC/Hooks

### PR DESCRIPTION
**Pull request by colinlohner**
## Overview 
Update TableRow to use hooks and be a function component
 
## Screenshots (if applicable) 
N/A

## Testing
- [x] Existing functionality remains the same

### CODE REVIEW

File: src/table/src/TableRow.js [hunk 1]
```diff
@@ -1,164 +1,149 @@
...
+        onSelect()
         e.preventDefault()
       } else if (key === 'ArrowUp' || key === 'ArrowDown') {
         try {
-          manageTableRowFocusInteraction(key, this.mainRef)
+          manageTableRowFocusInteraction(key, mainRef)
         } catch (_) {}
       } else if (key === 'Escape') {
-        if (this.mainRef) this.mainRef.blur()
+        if (mainRef && mainRef instanceof Node) mainRef.blur()
```
Comment (mshwery) [inline src/table/src/TableRow.js:60]: Why do we need `instanceof Node` here?

Comment (colinlohner) [reply src/table/src/TableRow.js:60]: just to make sure that whatever is passed back as the ref is an element, without it, it was failing to `blur`

File: src/table/src/TableRow.js [hunk 2]
```diff
@@ -1,164 +1,149 @@
...
-    this.props.onKeyPress(e)
+    onKeyPress(e)
   }
 
-  onRef = ref => {
-    this.mainRef = ref
-    safeInvoke(this.props.innerRef, ref)
+  const onRef = newRef => {
+    setMainRef(newRef)
+    safeInvoke(ref, newRef)
```
Comment (mshwery) [inline src/table/src/TableRow.js:69]: Does this work when the ref is from React.createRef or useRef?

Comment (mshwery) [reply src/table/src/TableRow.js:69]: tested https://codesandbox.io/s/reverent-frog-dlgzn?file=/src/App.js and it does not. We'll need to manually detect which kind of ref it is.

Comment (colinlohner) [reply src/table/src/TableRow.js:69]: Gonna do this in a separate PR to update across other components

**Pull request merged**